src/main/java/org/wildfly/security/auth/server/AbstractMechanismAuthenticationFactory.java

```diff
@@ -39,7 +39,7 @@
         this.mechanismConfigurations = mechanismConfigurations;
     }
 
-    SecurityDomain getSecurityDomain() {
+    public SecurityDomain getSecurityDomain() {
```

Comment:
I was actually planning on just referencing the HttpAuthenticationFactory and SaslAuthenticationFactory which could leave that interface purely about mechanisms.